monai/transforms/intensity/array.py

```
@@ -2161,3 +2157,97 @@ def __call__(self, img: torch.Tensor) -> torch.Tensor:
+        if threshold is not None:
+            if isinstance(threshold, dict):
+                for mode, th in threshold.items():
+                    self._set_threshold(th, mode.upper())
+            else:
+                self._set_threshold(threshold, "R")
+                self._set_threshold(threshold, "G")
+                self._set_threshold(threshold, "B")
+        if hsv_threshold is not None:
+            if isinstance(hsv_threshold, dict):
+                for mode, th in hsv_threshold.items():
+                    self._set_threshold(th, mode.upper())
+            else:
+                self._set_threshold(hsv_threshold, "H")
+                self._set_threshold(hsv_threshold, "S")
+                self._set_threshold(hsv_threshold, "V")
+
+        self.invert = invert
+
+    def _set_threshold(self, threshold, mode):
+        if callable(threshold):
+            self.thresholds[mode] = threshold
+        elif isinstance(threshold, str):
+            self.thresholds[mode] = getattr(skimage.filters, "threshold_" + threshold.lower())
+        elif isinstance(threshold, (float, int)):
+            self.thresholds[mode] = float(threshold)
+        else:
+            raise ValueError(
+                f"`threshold` should be either a callable, string, or float number, {type(threshold)} was given."
+            )
+
+    def _get_threshold(self, image, mode):
+        threshold = self.thresholds.get(mode)
+        if callable(threshold):
+            return threshold(image)
+        return threshold
+
+    def __call__(self, image: NdarrayOrTensor):
+        img_rgb, *_ = convert_data_type(image, np.ndarray)
+        if self.invert:
+            img_rgb = skimage.util.invert(img_rgb)
+
+        foregrounds = []
+        if set("RGB") & set(self.thresholds.keys()):
+            rgb_foreground = np.zeros_like(img_rgb[:1])
+            for img, mode in zip(img_rgb, "RGB"):
+                threshold = self._get_threshold(img, mode)
+                if threshold:
```

Should we allow `threshold==None` here? I think maybe we should raise an exception or warning?

Thanks.

---

Now it raises an exception if the `threshold=None` or wrongly set.

---

May I know where it raises an error now?
I mean currently, this function can return None: `_get_threshold`.

Thanks.

---

OK, I see this part. If no None case in `_get_threshold`, seems we can remove this `if threshold:` line at line 2242 and line 2250?

Thanks.